File: Práctica_01/CPP/EJ_01.cpp
#include <iostream>
#include "vector"
using namespace std;

//Con array
int suma_arr(int *arrreglo,int tamaño){
    int cont = 0;
    for(int i = 0; i < tamaño; i++){
        cont += arrreglo[i];
    }
    return cont;
}

//Con vector
int suma_arr_2(vector<int> vec){
    int cont = 0;
    for (int i : vec){
        cont += i;
    }
    return cont;
}

int main(){
    int arr[5] = {1,2,3,4,5};
    vector vec = {1,3,5,7};

    int rpta = suma_arr(arr,5);
    int rpta_2 = suma_arr_2(vec);

    cout << "La suma es " << rpta << endl;
    cout << "La suma usando vector es " << rpta_2 << endl;

    return 0;
};

File: Práctica_01/JAVA/EJ_01.java
import java.util.Vector;

public class EJ_01 {

    public static void main(String[] args) {

        Vector<Integer> numeros_2 = new Vector();

        for (int i = 0; i < 6; i++) {
            numeros_2.add(i);
        }


        int[] numeros = new int[6];
        for (int i = 0; i < 6; i++) {
            numeros[i] = i;
        }


        int rpta = suma_arr(numeros);
        int rpta_2 = suma_arr2(numeros_2);
        System.out.println("Suma con arreglo -> " + rpta);
        System.out.println("Suma vector -> " + rpta_2);

    }

    public static int suma_arr(int[] arreglo){
        int cont = 0;

        for (int i : arreglo){
            cont += i;
        }

        return cont;
    }

    public static int suma_arr2(Vector<Integer> vec){
        int cont = 0;
        for (int i : vec){
            cont += i;
        }
        return cont;
    }

}
